controllers/profiles/common/object_creators.go

```diff
@@ -51,14 +54,17 @@ import (
 
 // ObjectCreator is the func that creates the initial reference object, if the object doesn't exist in the cluster, this one is created.
 // Can be used as a reference to keep the object immutable
-type ObjectCreator func(workflow *operatorapi.SonataFlow) (client.Object, error)
+type ObjectCreator func(workflow *operatorapi.SonataFlow, support *StateSupport) (client.Object, error)
```

Comment:
Please review this API change, it's unnecessary if all you need is the `client`. That brings a lot of change to the code base. Same for the `Ensurer`.

Reply:
@ricardozanini Thank you for your suggestion. Looking into it.

Reply:
Done